axlearn/common/flash_attention/neuron_attention.py

```diff
@@ -0,0 +1,124 @@
+from functools import partial
+import jax
+import jax.numpy as jnp
+from jax import custom_vjp
+
+lnc = 2 if jax.devices()[0].device_kind == "NC_v3d" else 1
+
+@partial(custom_vjp, nondiff_argnums=(4, 5))
+def flash_attention(query, key, value, bias, causal, softmax_scale):
+    out, _ = _mha_forward(query, key, value, bias, causal, softmax_scale)
+    return out
+
+
+def _mha_forward(query, key, value, bias, causal, softmax_scale):
```

Comment:
Can we get a support for segment ID and dropout as well? Both are quite needed nowadays.